cluster-assign-roles.sh

```diff
@@ -181,10 +181,24 @@ function hadoop_install {
   done
 
   printf "Installing workers...\n"
+  status_file=${TMPDIR:-/tmp}/status_file_$$
+  function clean_up_status_file {
+    rm -f $status_file
+  }
+  trap clean_up_status_file EXIT
+  tempfile -n $status_file
```

Comment:
On the Ubuntu bootstrap node, with GNU bash, version 4.2.25(1)-release (x86_64-pc-linux-gnu), tempfile is available and the code does work.